### PR TITLE
Initial implementation of Twilio connector

### PR DESCRIPTION
**Pull request by ldclakmal**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: twilio/twillio_connector.bal [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            VoiceCallResponse voiceCallResponse = {};
+            voiceCallResponse.sid = jsonPayload.sid != null ? jsonPayload.sid.toString() : EMPTY_STRING;
+            voiceCallResponse.status = jsonPayload.status != null ? jsonPayload.status.toString() : EMPTY_STRING;
+            voiceCallResponse.price = jsonPayload.price != null ? jsonPayload.price.toString() : EMPTY_STRING;
+            voiceCallResponse.priceUnit = jsonPayload.price_unit != null ? jsonPayload.price_unit.toString() : EMPTY_STRING;
+            return voiceCallResponse;
+        }
+        error err => return err;
+    }
+}
```
[the remark this answers is not in the document]
Comment (ldclakmal) [reply twilio/twillio_connector.bal]: Fixed.

File: twilio/twillio_endpoint.bal [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    http:HttpClient httpClient = http:createHttpClient(twilioConfig.uri, twilioConfig.clientConfig);
+    twilioEndpoint.twilioConnector = {accountSid:twilioConfig.accountSid, authToken:twilioConfig.authToken};
+    twilioEndpoint.twilioConnector.clientEndpoint.httpClient = httpClient;
+}
+
+@Description {value:"Returns the connector that client code uses."}
+@Return {value:"The connector that client code uses."}
+public function <TwilioEndpoint twilioEndpoint> getClient () returns TwilioConnector {
+    return twilioEndpoint.twilioConnector;
+}
```
[the remark this answers is not in the document]
Comment (ldclakmal) [reply twilio/twillio_endpoint.bal]: Added the missing methods and fixed.

File: .gitignore [hunk 3]
```diff
@@ -20,3 +20,10 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+# IDE file
+.idea
+*.iml
+
+# Ballerina package files
+*.toml
```
[the remark this answers is not in the document]
Comment (ldclakmal) [reply .gitignore]: Fixed.

File: twilio/twilio_utils.bal [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// under the License.package twilio;
+
+package twilio;
+
+import ballerina/net.uri;
+import ballerina/util;
+
+@Description {value:"Create basic authorization header value with encoded account sid and auth token."}
+@Return {value:"Encoded header value."}
+public function <TwilioConnector twilioConnector> getAuthorizationHeaderValue () returns string {
```
Comment (muthulee) [inline twilio/twilio_utils.bal]: Move to private please

**Pull request merged**